ambry-store/src/main/java/com/github/ambry/store/BlobStoreCompactor.java

```diff
@@ -821,19 +824,29 @@ private boolean copyRecords(LogSegment logSegmentToCopy, List<IndexEntry> srcInd
           long usedCapacity = tgtIndex.getLogUsedCapacity();
           if (isActive && (tgtLog.getCapacityInBytes() - usedCapacity >= srcValue.getSize())) {
             Offset endOffsetOfLastMessage = tgtLog.getEndOffset();
-            if (config.storeCompactionMinOperationsBytesPerSec < config.storeCompactionOperationsBytesPerSec) {
+            if (config.storeCompactionMinOperationsBytesPerSec < config.storeCompactionOperationsBytesPerSec
+                && srcMetrics.diskReadTimePerMbInMs.containsKey(diskMountPath)
```

Comment:
Could the disk metrics be saved as class members in the constructor of `BlobStoreCompactor`? `diskMountPath` never changes so maybe we can avoid the hashmap lookup each entry